runtime/README.md

```diff
@@ -0,0 +1,488 @@
+for use.
+
+### Fault Handling
+
+A mailbox command can fail to complete in a couple ways
+
+* Hang/timeout which results in the watchdog firing
+* Unrecoverable panic
+
+In both these cases, the panic handler will write diagnostic panic information
+to registers readable by the SoC, firmware will undergo impactless reset, and
+`mailbox_data_avail` will be asserted.
+
+The caller is expected to check status registers upon reading responses from the
+mailbox.
+
+Depending on the type of fault, the SoC may:
+
+* Resubmit the mailbox command
+* Attempt to update Runtime Firmware
+* Perform a full SoC reset
+* Some other SoC-specific behavior
+
+### Drivers
+
+Caliptra Runtime Firmware will share driver code with ROM and FMC where
+possible, however it will have its own copies of all these drivers linked into
+the Runtime Firmware binary.
+
+## Maibox Commands
+
+All mailbox command codes are little endian.
+
+Table: Mailbox command "result" codes:
+
+| **Name**         | **Value**              | Description
+| -------          | -----                  | -----------
+| `SUCCESS`        | `0x0000_0000`          | Mailbox command succeeded
+| `BAD_VENDOR_SIG` | `0x5653_4947` ("VSIG") | Vendor signature check failed
+| `BAD_OWNER_SIG`  | `0x4F53_4947` ("OSIG") | Owner signature check failed
+| `BAD_SIG`        | `0x4253_4947` ("BSIG") | Generic signature check failure (for crypto offload)
+| `BAD_IMAGE`      | `0x4249_4D47` ("BIMG") | Malformed input image
+| `BAD_CRC`        | `0x4243_5243` ("BCRC") | CRC check failed on input arguments
+
+Relevant Mailbox Registers:
+
+* COMMAND: Command code to execute
+* DLEN: Number of bytes written to mailbox
+
+### CALIPTRA\_FW\_LOAD
+
+The `CALIPTRA_FW_LOAD` command is handled by both ROM and Runtime Firmware.
+
+#### ROM Behavior
+
+On cold boot, ROM will expose the `CALIPTRA_FW_LOAD` mailbox command to accept
+the firmware image that ROM will boot. This includes Manifest, FMC, and Runtime
+firmware.
+
+#### Runtime Firmware Behavior
+
+Caliptra Runtime FW will also expose this mailbox command for loading
+impactless updates. See “Runtime Firmware Updates” for details.
+
+Command Code: `0x4657_4C44` ("FWLD")
+
+Table: `CALIPTRA_FW_LOAD` input arguments
+
+| **Name**  | **Type**      | **Description**
+| --------  | --------      | ---------------
+| data      | u8[...]       | Firmware image to load.
+
+Table: `CALIPTRA_FW_LOAD` output arguments
+
+| **Name** | **Type** | **Description**
+| -------- | -------- | ---------------
+| crc      | u32      | CRC over other output arguments, computed by Caliptra. Little endian.
+| result   | u32      | Result code. Little endian.
+
+### GET\_IDEV\_CSR
+
+ROM exposes a command to get a self-signed IDEVID CSR.
+GET\_IDEV\_CSR is not exposed by runtime firmware.
+
+Command Code: `0x4944_4556` ("IDEV")
+
+`GET_IDEV_CSR` takes no input arguments.
+
+* When invoked via JTAG, ROM will write the CSR to the mailbox so it can
```

Comment:
Is this statement meaningful? The outcome is the same whether using JTAG or not.

Reply:
Agreed, removed.